library/src/main/java/com/instacart/library/truetime/CacheInterface.java

```diff
@@ -0,0 +1,27 @@
+package com.instacart.library.truetime;
+
+import android.support.annotation.StringDef;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+
+public interface CacheInterface {
+
+    String KEY_CACHED_BOOT_TIME = "com.instacart.library.truetime.cached_boot_time";
+    String KEY_CACHED_DEVICE_UPTIME = "com.instacart.library.truetime.cached_device_uptime";
+    String KEY_CACHED_SNTP_TIME = "com.instacart.library.truetime.cached_sntp_time";
+
+    void put(@TrueTimeCacheType String key, long value);
+
+    long get(@TrueTimeCacheType String key, long defaultValue);
+
+    void clear();
+
+    @StringDef({KEY_CACHED_BOOT_TIME,
```

Comment:
hmm is there an explicit benefit. I ask cause removing this would basically allow us to avoid the `compile "com.android.support.X.."` dependency as well ?

Reply:
Yes, it will remove the annotation dependency. I actually had thought about it at the very beginning when I was working on these support. And I recalled my justification is because the `com.android.support:appcompat-v7 ` would also include the annotation dependency and I also checked the source code of the annotation package, which is really few, so I added it. I just want to bring the library users attention that this is an interface only intended for saving three type of data. No more. I will be happy to remove it if you insist.

Reply:
yeah i'd prefer if we didn't have it (maybe we just create an enum with those three types to control the inputs apis). i'm totally fine with `enum`s :)